### PR TITLE
Upgrade Confluent Versions to 3.3.1, and 4.0.0.

### PR DESCRIPTION
**Pull request by cyan-confluent**
Deprecate all versions below 3.3.1.

### CODE REVIEW

**Comment by dbtucker** [comment]
Best to update the default in nodegroup.template as well (even though it's never invoked on it's own).

Otherwise, LGTM.

**Comment by cyan-confluent** [comment]
Will do.

File: scripts/cp-deploy.sh [hunk 1]
```diff
@@ -514,10 +514,12 @@ configure_workers() {
 	if [ $? -eq 0 ] ; then
 		if [ -d $CP_HOME/share/java ] ; then
 			KC_PLUGIN_DIR=${CP_HOME}/share/java/kc-plugins
+			KC_PLUGIN_PATH=$CP_HOME/share/java,$KC_PLUGIN_DIR
 		else
 			KC_PLUGIN_DIR=/usr/share/java/kc-plugins
+			KC_PLUGIN_PATH=/usr/share/java,$KC_PLUGIN_DIR
 		fi
-		mkdir $KC_PLUGIN_DIR
+        mkdir -p $KC_PLUGIN_DIR
```
Comment (santiagocardenas) [inline scripts/cp-deploy.sh]: Tabs seem off?

**Comment by santiagocardenas** [comment]
We also must update the ci file that uses 3.1.2 to use 4.0.0 instead (that is, the non-default version) by changing, the parameter value, file name, and reference to file name in config.yml.

**Pull request merged**